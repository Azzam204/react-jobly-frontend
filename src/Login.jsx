import React, { useContext, useState } from "react";
import {
  Card,
  CardBody,
  FormGroup,
  Form,
  Label,
  Input,
  Button
} from 'reactstrap';
import { useNavigate } from 'react-router-dom'

/** Login page
 * 
 * Displays log in form
 * 
 * calls handleLogin on submit
 * 
 * if successfull, redirects to home
 * 
 * else, display errors
 */ 


function Login({ handleLogin }) {

  const DEFAULT = {
    username: '',
    password: ''
  }

  const navigate = useNavigate()

  const [formData, setFormData] = useState(DEFAULT)
  const [error, setError] = useState(false)

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data, [name]: value
    }))
  }

  async function handleSubmit(e) {
    e.preventDefault();
    let res = await handleLogin(formData)
    res.error ? setError(res.error) : navigate('/')
  }

  return (
    <div className="pt-5 container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
      <h3 className="mb-3 white" >
        Log In
      </h3>
      <Card >
        <CardBody>
          <Form onSubmit={handleSubmit}>
            <FormGroup className="mb-3">
              <Label for='username' className="fw-bold">
                Username
              </Label>
              <Input
                id="username"
                name="username"
                type="text"
                value={formData.username}
                onChange={handleChange}
                required
              />

              <Label for='password' className="fw-bold">
                Password
              </Label>
              <Input
                id="password"
                name="password"
                type="password"
                value={formData.password}
                autoComplete="on"
                onChange={handleChange}
                required
              />

              {error && <div className="alert alert-danger" role="alert">{error}</div>}

              <div className="d-grid gap-2 mt-3">
                <Button color="primary">Submit</Button>
              </div>
            </FormGroup>
          </Form>
        </CardBody>
      </Card>
    </div>
  )
}

export default Login;