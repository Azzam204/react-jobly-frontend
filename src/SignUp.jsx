import React, { useState } from "react";
import {
  Card,
  CardBody,
  FormGroup,
  Form,
  Label,
  Input,
  Button
} from 'reactstrap';
import { useNavigate } from 'react-router-dom'

/** Sign up page
 * 
 * Displays sign up form
 * 
 * calls handleSignup on submit
 * 
 * if successfull, redirects to home
 * 
 * else, display errors
 */ 


function SignUp({ handleSignUp }) {

  const DEFAULT = {
    username: '',
    password: '',
    firstName: '',
    lastName: '',
    email: ''
  }

  const navigate = useNavigate()

  const [formData, setFormData] = useState(DEFAULT)
  const [error,setError] = useState(false)

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data, [name]: value
    }))
  }

  async function handleSubmit(e) {
    e.preventDefault();
    let res = await handleSignUp(formData)
    setFormData(DEFAULT)
    res.error ? setError(res.error) : navigate('/')
  }

  return (
    <div className="pt-5 container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
      <h3 className="mb-3 white" >
        Sign Up
      </h3>
      <Card >
        <CardBody>
          <Form onSubmit={handleSubmit}>
            <FormGroup className="mb-3">
              <Label for='username' className="fw-bold">
                Username
              </Label>
              <Input
                id="username"
                name="username"
                type="text"
                value={formData.username}
                onChange={handleChange}
                required
              />

              <Label for='password' className="fw-bold">
                Password
              </Label>
              <Input
                id="password"
                name="password"
                type="password"
                value={formData.password}
                autoComplete="on"
                onChange={handleChange}
                required
              />

              <Label for='firstName' className="fw-bold">
                First name
              </Label>
              <Input
                id="firstName"
                name="firstName"
                type="text"
                value={formData.firstName}
                onChange={handleChange}
                required
              />

              <Label for='lastName' className="fw-bold">
                Last name
              </Label>
              <Input
                id="lastName"
                name="lastName"
                type="text"
                value={formData.lastName}
                onChange={handleChange}
                required
              />

              <Label for='email' className="fw-bold">
                Email
              </Label>
              <Input
                id="email"
                name="email"
                type="email"
                value={formData.email}
                onChange={handleChange}
                required
              />

              {error && <div className="alert alert-danger" role="alert">{error}</div>}

              <div className="d-grid gap-2 mt-3">
                <Button color="primary">Submit</Button>
              </div>
            </FormGroup>
          </Form>
        </CardBody>
      </Card>
    </div>
  )
}

export default SignUp;