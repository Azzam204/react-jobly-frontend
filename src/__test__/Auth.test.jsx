import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Login from "../Login";
import SignUp from "../SignUp";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <Login />
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <Login />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <SignUp />
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <SignUp />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});