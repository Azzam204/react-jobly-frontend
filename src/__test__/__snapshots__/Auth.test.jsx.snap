// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-5 container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
  >
    <h3
      class="mb-3 white"
    >
      Log In
    </h3>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3 mb-3"
          >
            <label
              class="fw-bold form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
            <label
              class="fw-bold form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="on"
              class="form-control"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="pt-5 container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
  >
    <h3
      class="mb-3 white"
    >
      Sign Up
    </h3>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3 mb-3"
          >
            <label
              class="fw-bold form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
            <label
              class="fw-bold form-label"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="on"
              class="form-control"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <label
              class="fw-bold form-label"
              for="firstName"
            >
              First name
            </label>
            <input
              class="form-control"
              id="firstName"
              name="firstName"
              required=""
              type="text"
              value=""
            />
            <label
              class="fw-bold form-label"
              for="lastName"
            >
              Last name
            </label>
            <input
              class="form-control"
              id="lastName"
              name="lastName"
              required=""
              type="text"
              value=""
            />
            <label
              class="fw-bold form-label"
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              required=""
              type="email"
              value=""
            />
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
